lib/moab/storage_object_validator.rb

```diff
@@ -88,18 +98,36 @@ def check_correctly_formed_moabs
       version_directories.each do |version_dir|
         version_path = "#{storage_obj_path}/#{version_dir}"
         version_sub_dirs = sub_dirs(version_path)
-        before_result_size = errors.size
+        version_error_count = errors.size
         errors.concat check_sub_dirs(version_sub_dirs, version_dir, EXPECTED_VERSION_SUB_DIRS)
-        after_result_size = errors.size
         # run the following checks if this version dir passes check_sub_dirs, even if some prior version dirs didn't
-        if before_result_size == after_result_size
-          data_dir_path = "#{version_path}/#{DATA_DIR_NAME}"
-          data_sub_dirs = sub_dirs(data_dir_path)
-          errors.concat check_sub_dirs(data_sub_dirs, version_dir, EXPECTED_DATA_SUB_DIRS)
-          errors.concat check_required_manifest_files(version_path, version_dir)
+        errors.concat check_required_manifest_files(version_path, version_dir) if version_error_count == errors.size
+        if version_error_count == errors.size
+          errors.concat check_expected_data_sub_dirs(version_path, version_dir)
+          if version_error_count == errors.size
+            errors.concat check_metadata_dir_files_only(version_path)
+          end
```

Comment:
I'm okay leaving this, but it just occurred to me:  can we now do:

```
version_dir_error_count = errors.size
errors.concat check_sub_dirs(version_sub_dirs, version_dir, EXPECTED_VERSION_SUB_DIRS)
errors.concat check_required_manifest_files(version_path, version_dir) if version_dir_error_count == errors.size
errors.concat check_expected_data_sub_dirs(version_path, version_dir) if version_dir_error_count == errors.size
errors.concat check_metadata_dir_files_only(version_path) if version_dir_error_count == errors.size
```

Is that clearer or less clear?  I'm ok if you want to make a separate refactor ticket for it.

Reply:
i like it. makes it more clear!